storage/innobase/log/log0online.cc

```diff
@@ -1350,13 +1350,34 @@ bool log_online_follow_redo_log(void) {
   if (result) {
     result = log_online_write_bitmap();
     log_bmp_sys->start_lsn = log_bmp_sys->end_lsn;
-    log_sys->tracked_lsn.store(log_bmp_sys->start_lsn);
+    log_sys->tracked_lsn.store(log_bmp_sys->parse_buf.get_current_lsn());
   }
 
   mutex_exit(&log_bmp_sys_mutex);
   return result;
 }
 
+/** Read and parse redo log past the last checkpoint lsn for FLUSH
+CHANGED_PAGE_BITMAPS command
+
+@return true if log tracking succeeded, false if bitmap write I/O error
+*/
+bool log_online_follow_redo_log_past_checkpoint_lsn(void) {
+  ut_ad(!srv_read_only_mode);
+
+  const auto last_checkpoint_lsn = log_get_checkpoint_lsn(*log_sys);
+  bool result = true;
+
+  DEBUG_SYNC_C("log_online_follow_redo_log_past_checkpoint_lsn");
+
+  while (result && srv_track_changed_pages &&
+         last_checkpoint_lsn > log_sys->tracked_lsn.load()) {
+    result = log_online_follow_redo_log();
```

Comment:
if (!result) break?

Reply:
`if (!result) break'` is not needed, because the loop is `while (result)`

Reply:
Right, sorry